extra/innochecksum.cc

```diff
@@ -1428,6 +1448,85 @@ rewrite_checksum(
 		&& !write_file(filename, fil_in, buf, flags, pos);
 }
 
+/** Read and validate page 0, then initialize tablespace flags
+and page size.
+@param	fil_in        File pointer
+@param	buf           Buffer to read page into
+@return whether the page was read successfully */
+static bool read_and_validate_page0(FILE *fil_in, byte *buf)
+{
+  /* Read the minimum page size first */
+  bool use_tablespace_flags= (tablespace_flags != ~0U);
+  if (use_tablespace_flags)
+    init_page_size_from_flags(tablespace_flags);
+  else
+    /* Initialize with default page size first, will be updated after reading page 0 */
+    init_page_size_from_flags(0);
```

Comment:
Can we introdude a `constexpr uint32_t USE_FSP_FLAGS{UINT32_MAX};` and use it everywhere instead of `UINT32_MAX` or `~0U`?

Could we avoid duplicating the function call?
```c++
  init_page_size_from_flags(tablespace_flags == USE_FSP_FLAGS
                            ? 0 : tablespace_flags);
```